cmd/ssh.go

```diff
@@ -18,141 +19,47 @@ import (
-		fmt.Sprintf("tar -xzf %s -C %s && rm -rf %s", tmpDstPath, dstPath, tmpDstPath),
-	)
-
-	sshCommand.Stdout = stdout
-	sshCommand.Stderr = stderr
-
-	if err := sshCommand.Run(); err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			// Exited with non-zero exit code
-			if status, ok := exitError.Sys().(syscall.WaitStatus); ok {
-				if status.ExitStatus() == 255 {
-					ui.Infof("The remote host closed the connection.")
-					return fmt.Errorf("failed to copy source: %w", err)
-				}
-			}
-			ui.Infof("Failed to extract source directory on remote host: %s", stderr.String())
-			return err
-		}
-		return fmt.Errorf("failed to unzip on remote host: %v", err)
-	}
-
-	ui.Infof("✅  Successfully copied source directory to remote host")
-
-	return nil
+// SSH handles the Cobra command for SSH
+func SSH(cmd *cobra.Command, args []string) error {
+	return sessionConnect(cmd, false, args)
 }
 
-func SSH(cmd *cobra.Command, args []string) error {
+// sessionConnect handles the flow to spawn a new SSH connection to an exec.
+func sessionConnect(cmd *cobra.Command, withOpenVSCode bool, args []string) error {
 	cmd.SilenceUsage = true
 	ctx := cmd.Context()
 
-	// TODO: parse args and forward to ssh command
+	sshArgsByKey, err := parseSSHArgs(args)
+	if err != nil {
+		const errMsg = "❌ Invalid arguments. If you want to pass arguments to the ssh command, " +
+			"use the -- flag. See `unweave ssh --help` for more information"
+		ui.Errorf(errMsg)
+		os.Exit(1)
+	}
 
-	errMsg := "❌ Invalid arguments. If you want to pass arguments to the ssh command, " +
-		"use the -- flag. See `unweave ssh --help` for  more information."
+	// Lays the ground to fix the following:
+	// TODO Fix a bug where unweave ssh {exec name / exec id} was not implemented/working
+	execRef, err := getExecRefFromArgs(cmd.Context(), sshArgsByKey)
+	if err != nil {
+		ui.Errorf(err.Error())
+		os.Exit(1)
+	}
 
 	execCh := make(chan types.Exec)
 	errCh := make(chan error)
 	isNew := false
 
-	var err error
-	var sshArgs []string
-	var execRef string // Can be execID or name
-
-	// If the number of args is great than one, we always expect the first arg to be
-	// the separator flag "--". If the number of args is one, we expect it to be the
-	// execID or name
-	if len(args) > 1 {
-		sshArgs = args[1:]
-		if sshArgs[0] != "--" {
-			ui.Errorf(errMsg)
-			os.Exit(1)
-		}
-
-		if len(args) == 1 {
-			execRef = args[0]
-		} else {
-			execRef = args[0]
-		}
-	}
-
 	watchCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	if config.CreateExec {
-		// If the flag to create a new exec is passed, any arguments must be forwarded to
-		// the ssh command
-		if len(args) > 0 && args[0] != "--" {
-			ui.Errorf(errMsg)
-			os.Exit(1)
-		}
-
 		ui.Infof("Initializing node...")
 
 		execCh, errCh, err = execCreateAndWatch(watchCtx, types.ExecConfig{}, types.GitConfig{})
 		if err != nil {
 			return err
 		}
 		isNew = true
-
 	} else {
 		var createNewExec bool
 		if execRef == "" {
```

Comment:
`sessionConnect` is the main function that DRYs up the VSCode and SSH flows. It just decides how you get your exec and SSH settings, then passes on the result to the SSH or VSCode flows respectively.